src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +30,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount = -numberOfTorpedos;
```

Comment:
Én ezt nem így oldottam meg, úgy értelmeztem, hogy a -= és az =- is ugyanúgy fordul, viszont te ezt pont arra írtad át, amit el szerettünk volna kerülni. Annak nincs sok értelme, hogy a torpedó számlálót átállítsuk a torpedók számának negáltjára, hanem kivonjuk belőle, amiket épp kilőttünk.